deeposlandia/postprocess.py

```diff
@@ -0,0 +1,440 @@
+                        help="Relative path towards data directory")
+    parser.add_argument('-s', '--image-size',
+                        required=True, type=int,
+                        help="Image patch size, in pixels")
+    return parser
+
+
+def get_image_paths(datapath, dataset, image_size, image_basename):
+    """Returns a list with the path of every image that belongs to the
+    `dataset`, preprocessed in `image_size`-pixelled images, that were
+    extracted from an original image named as `image_basename`.
+
+    Parameters
+    ----------
+    datapath : str
+        Path of the data on the file system
+    dataset : str
+        Name of the dataset
+    image_size : int
+        Size of preprocessed images, in pixels
+    image_basename : str
+        Original image name
+
+    Returns
+    -------
+    list
+        List of image full paths
+    """
+    image_raw_paths = os.path.join(
+        datapath, dataset, "preprocessed", str(image_size) + "_full",
+        "testing", "images", image_basename + "*"
+    )
+    return [glob.glob(f) for f in [image_raw_paths]][0]
+
+
+def extract_images(image_paths):
+    """Convert a list of image filenames into a numpy array that contains the
+    image data
+
+    Parameters
+    ----------
+    image_paths : str
+        Name of the image files onto the file system
+
+    Returns
+    -------
+    np.array
+        Data that is contained into the image
+    """
+    x_test = []
+    for image_path in image_paths:
+        if not image_path.endswith(".png"):
+            logger.error("The filename does not refer to an png image.")
+            raise ValueError()
+        image = Image.open(image_path)
+        if image.size[0] != image.size[1]:
+            logger.error(
+                "One of the parsed images has non-squared dimensions."
+            )
+            raise ValueError()
+        x_test.append(np.array(image))
+    return np.array(x_test)
+
+
+def get_labels(datapath, dataset, tile_size):
+    """Extract labels from the `dataset` glossary, according to the
+    preprocessed version of the dataset
+
+    Parameters
+    ----------
+    datapath : str
+        Path of the data on the file system
+    dataset : str
+        Name of the dataset
+    tile_size : int
+        Size of preprocessed images, in pixels
+
+    Returns
+    -------
+    list
+        List of dictionnaries that describes the dataset labels
+    """
+    prepro_folder = utils.prepare_preprocessed_folder(
+        datapath, dataset, tile_size, "full"
+    )
+    if os.path.isfile(prepro_folder["testing_config"]):
+        test_config = utils.read_config(prepro_folder["testing_config"])
+    else:
+        raise ValueError(("There is no testing data with the given "
+                          "parameters. Please generate a valid dataset "
+                          "before calling the program."))
+    labels = [l for l in test_config["labels"] if l["is_evaluate"]]
```

Comment:
no need to create a variable. I think you can return the list directly

Reply:
Simplified!